packages/appcache-polyfill-window/index.ts

```diff
@@ -0,0 +1,222 @@
+    };
+    const request = new Request(url, init);
+
+    try {
+      const response = await fetch(request);
+
+      const cacheControl = response.headers.get('Cache-Control');
+      if (cacheControl && cacheControl.includes('no-store')) {
+        // Bail early if we're told not to cache this response.
+        return;
+      }
+
+      if (response.status === 200) {
+        await cache.put(url, response);
+        return;
+      }
+
+      // See Item 18.5 of https://html.spec.whatwg.org/multipage/browsers.html#downloading-or-updating-an-application-cache
+      if (response.status !== 404 && response.status !== 410) {
+        // Assuming this isn't a 200, 404 or 410, we want the catch to
+        // trigger, which will cause any previously cached Response for this
+        // URL to be copied over to this new cache.
+        throw new Error(`Bad status: ${response.status}`);
+      }
+    } catch (e) {
+      // We're here if one of the following happens:
+      // - The fetch() rejected due to a NetworkError.
+      // - The HTTP status code from the fetch() was something other than
+      //   200, 404, and 410 AND the response isn't Cache-Control: no-store
+      const response = await caches.match(url);
+      if (response) {
+        await cache.put(url, response);
+      }
+    }
+  }));
+}
+
+async function checkManifestVersion(manifestUrl: string) {
+  // See Item 4 of https://html.spec.whatwg.org/multipage/browsers.html#downloading-or-updating-an-application-cache
+  const init: RequestInit = {
+    credentials: 'include',
+    headers: [['X-Use-Fetch', 'true']],
+  };
+  const manifestRequest = new Request(manifestUrl, init);
+
+  let manifestResponse = await fetch(manifestRequest);
+  // See Item 6 of https://html.spec.whatwg.org/multipage/offline.html#downloading-or-updating-an-application-cache
+  if (manifestResponse.status === 404 || manifestResponse.status === 410) {
+    // TODO: Double check that this is a reasonable approach.
+    // If the manifest has explicitly been removed, then remove our saved state
+    // related to that manifest, but don't delete anything from the
+    // Cache Storage API. Developers migrating off of App Cache and to a full
+    // service worker might find themselves in this state.
+    return null;
+  }
+
+  // See https://html.spec.whatwg.org/multipage/offline.html#cache-failure-steps
+  if (manifestResponse.status !== 200) {
+    // TODO: Double check that this is a reasonable approach.
+    // For non-200/404/410 responses, just abort.
+    throw new Error('Manifest response status: ${manifestResponse.status}.');
+  }
+
+  const dateHeaderValue = manifestResponse.headers.get('date');
+  if (dateHeaderValue) {
+    const manifestDate = new Date(dateHeaderValue).valueOf();
+    // Calculate the age of the manifest in milliseconds.
+    const manifestAgeInMillis = Date.now() - manifestDate;
+
+    // If the age is greater than 24 hours, then we need to fetch without
```

Comment:
This was something that we decided on back in the day, to prevent folks from getting stuck on a cached App Cache version.